yb-voyager/src/utils/gcs/gcsutils.go

```diff
@@ -0,0 +1,123 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package gcs
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/url"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+var client *storage.Client
+
+func createClientIfNotExists() {
+	if client != nil {
+		return
+	}
+	// Creates a client with default credentials.
+	c, err := storage.NewClient(context.Background())
+	if err != nil {
+		utils.ErrExit("create gcs client: %w", err)
+	}
+	client = c
+
+}
+
+func ValidateObjectURL(datadir string) error {
+	u, err := url.Parse(datadir)
+	if err != nil {
+		return fmt.Errorf("parsing the object of %q: %w", datadir, err)
+	}
+	bucket := u.Host
+	if bucket == "" {
+		return fmt.Errorf("missing bucket in gcs url %v", datadir)
+	}
+	return nil
+}
+
+func splitObjectPath(objectPath string) (string, string, error) {
+	u, err := url.Parse(objectPath)
+	if err != nil {
+		return "", "", err
+	}
+	bucket := u.Host
+	key := u.Path[1:] //remove initial "/", unable to find object with it
+	if bucket == "" {
+		return "", "", fmt.Errorf("missing bucket in gcs url %v", objectPath)
+	}
+	if key == "" {
+		return "", "", fmt.Errorf("missing key in gcs url %v", objectPath)
+	}
+	return bucket, key, nil
+}
+
+func ListAllObjects(url *url.URL) ([]string, error) {
+	createClientIfNotExists()
+	bucket := url.Host
+	key := ""
+	query := &storage.Query{}
+	if url.Path != "" {
+		key = url.Path[1:] //remove initial "/", unable to find object with it
+		query = &storage.Query{Prefix: key}
+	}
+	objectIter := client.Bucket(bucket).Objects(context.Background(), query)
+	var objectNames []string
+	for {
+		attrs, err := objectIter.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return objectNames, fmt.Errorf("Bucket(%q).Objects: %w", bucket, err)
+		}
+		objectNames = append(objectNames, attrs.Name)
```

Comment:
Is the `attrs.Name` a full path i.e. starting from `gs://`?

Reply:
No, it is with prefix/key included. I have already added a commit to remove that key from the objectName as Shubham reported one issue.